codemods/paragraphsInsideTableCells.js

```diff
@@ -0,0 +1,52 @@
+const visit = require('unist-util-visit');
+const visitParents = require('unist-util-visit-parents');
+const {
+  flatten,
+  hasAttribute,
+  isMdxBlockElement,
+  isMdxElement,
+  hasOnlyChild,
+  removeAttribute,
+  removeChild,
+} = require('./utils/mdxast');
+
+const paragraphsInsideTableCells = () => (tree) => {
+  visit(tree, isMdxBlockElement('td'), (td) => {
+    if (
+      hasOnlyChild('paragraph', td) &&
+      td.children[0].children.every(isMdxElement('p'))
+    ) {
+      flatten(td);
+    }
+
+    visitParents(
+      td,
+      (node) =>
+        isMdxElement('p', node) && hasAttribute('data-fixnewline', node),
+      (p, ancestors) => {
+        const parent = ancestors[ancestors.length - 1];
+        const list = ancestors.find((node) => node.type === 'list');
+        const listParent = ancestors[ancestors.indexOf(list) - 1];
+        const listIdx = listParent.children.indexOf(list);
+
+        listParent.children = [
+          ...listParent.children.slice(0, listIdx + 1),
+          p,
+          ...listParent.children.slice(listIdx + 1),
+        ];
+
+        removeAttribute('data-fixnewline', p);
+        removeChild(p, parent);
+      }
+    );
+
+    visit(td, isMdxElement('p'), (p) => {
+      p.type = 'paragraph';
+
+      delete p.name;
```

Comment:
what is happening exactly here? why are the name and attributes being deleted?

Reply:
As per the MDAST's interface for a [`Paragraph`](https://github.com/syntax-tree/mdast#paragraph) type, it only includes the `type` and `children` properties. Since I'm converting a JSX `<p>` tag to a markdown `paragraph`, I figured it best to remove the additional attributes from the AST node.

Likely it wouldn't hurt anything to leave it, but for completeness sake, I removed them to match the `Paragraph` interface.